utils/src/main/java/com/cloud/utils/net/NetworkProtocols.java

```diff
@@ -0,0 +1,362 @@
+        ProtocolNumbers.add(new ProtocolNumber(113, "PGM", "PGM Reliable Transport Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(114, "Any 0-hop protocol", "Any 0-hop protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(115, "L2TP", "Layer Two Tunneling Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(116, "DDX", "D-II Data Exchange (DDX)"));
+        ProtocolNumbers.add(new ProtocolNumber(117, "IATP", "Interactive Agent Transfer Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(118, "STP", "Schedule Transfer Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(119, "SRP", "SpectraLink Radio Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(120, "UTI", "UTI"));
+        ProtocolNumbers.add(new ProtocolNumber(121, "SMP", "Simple Message Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(122, "SM (deprecated)", "Simple Multicast Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(123, "PTP", "Performance Transparency Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(124, "ISIS over IPv4", ""));
+        ProtocolNumbers.add(new ProtocolNumber(125, "FIRE", ""));
+        ProtocolNumbers.add(new ProtocolNumber(126, "CRTP", "Combat Radio Transport Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(127, "CRUDP", "Combat Radio User Datagram"));
+        ProtocolNumbers.add(new ProtocolNumber(128, "SSCOPMCE", ""));
+        ProtocolNumbers.add(new ProtocolNumber(129, "IPLT", ""));
+        ProtocolNumbers.add(new ProtocolNumber(130, "SPS", "Secure Packet Shield"));
+        ProtocolNumbers.add(new ProtocolNumber(131, "PIPE", "Private IP Encapsulation within IP"));
+        ProtocolNumbers.add(new ProtocolNumber(132, "SCTP", "Stream Control Transmission Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(133, "FC", "Fibre Channel"));
+        ProtocolNumbers.add(new ProtocolNumber(134, "RSVP-E2E-IGNORE", ""));
+        ProtocolNumbers.add(new ProtocolNumber(135, "Mobility Header", ""));
+        ProtocolNumbers.add(new ProtocolNumber(136, "UDPLite", ""));
+        ProtocolNumbers.add(new ProtocolNumber(137, "MPLS-in-IP", ""));
+        ProtocolNumbers.add(new ProtocolNumber(138, "manet", "MANET Protocols"));
+        ProtocolNumbers.add(new ProtocolNumber(139, "HIP", "Host Identity Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(140, "Shim6", "Shim6 Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(141, "WESP", "Wrapped Encapsulating Security Payload"));
+        ProtocolNumbers.add(new ProtocolNumber(142, "ROHC", "Robust Header Compression"));
+        ProtocolNumbers.add(new ProtocolNumber(143, "Ethernet", "Ethernet"));
+        ProtocolNumbers.add(new ProtocolNumber(144, "AGGFRAG", "AGGFRAG encapsulation payload for ESP"));
+        ProtocolNumbers.add(new ProtocolNumber(145, "NSH", "Network Service Header"));
+    }
+    /**
+     * Different types of ICMP.
```

Comment:
is this comment correct? It seems any protocol could be encoded not just ICMP

Reply:
thanks @DaanHoogland for the review

the comment needs to be put to line 222 I think. I will make some other changes on the comments